packages/babel-preset-typescript/src/plugin-rewrite-ts-imports.ts

```diff
@@ -14,12 +24,18 @@ export default declare(function ({ types: t }) {
         const kind = t.isImportDeclaration(node)
           ? node.importKind
           : node.exportKind;
-        if (kind === "value" && source && /[\\/]/.test(source.value)) {
-          source.value = source.value
-            .replace(/(\.[mc]?)ts$/, "$1js")
-            .replace(/\.tsx$/, ".js");
+        if (kind === "value" && source) {
+          maybeReplace(source);
         }
       },
+      CallExpression(path) {
+        if (t.isImport(path.node.callee)) {
+          maybeReplace(path.node.arguments[0]);
+        }
+      },
```

Comment:
Can we mark the CallExpression visitor Babel 7 only?

Reply:
We haven't removed `createImportExpressions` in Babel 8, so users can still use it. :)